source/concurrent/Promise.ooc

```diff
@@ -131,19 +156,27 @@ _ThreadFuture: class <T> extends Future<T> {
 		if (this _threadAlive)
 			if (this _thread wait(time toSeconds()))
 				this _threadAlive = false
-		this _state == _PromiseState Finished
+		this _stateMutex lock()
+		result := this _state == _PromiseState Finished
+		this _stateMutex unlock()
+		result
 	}
 	getResult: override func (defaultValue: T) -> T {
 		result := defaultValue
```

Comment:
No, looks like it's just a memcpy with a constant size:
```ooc
Test: class <T> {
    init: func
    fun: func (value: T) {
		x := value
    }
}


void prob__Test_fun(prob__Test* this, uint8_t* value) {
    uint8_t* x = Memory__alloca(this->T->size);
    Memory__memcpy(x, value, this->T->size);
}
```
no "class constructor" calls here.